src/main/java/no/ndla/taxonomy/repositories/NodeRepository.java

```diff
@@ -43,10 +43,10 @@ public interface NodeRepository extends TaxonomyRepository<Node> {
             LEFT JOIN n.parentConnections pc
             WHERE ((:#{#nodeTypes == null} = true) OR n.nodeType in (:nodeTypes))
             AND (:isVisible IS NULL OR n.visible = :isVisible)
-            AND (:metadataFilterKey IS NULL OR jsonb_extract_path_text(n.customfields, :metadataFilterKey) IS NOT NULL)
+            AND (:metadataFilterKey IS NULL OR jsonb_extract_path_text(n.customfields, cast(:metadataFilterKey as text)) IS NOT NULL)
```

Comment:
Er disse cast'ene fordi det er optionals?

Reply:
Usikker. Hibernate klaga på at det var sammenligning av tekst og object men ved å legge til cast så funka det. Samme med json_contains i linja under.

Reply:
Det skremmer meg jo litt, men så lenge det fungerer :shrug:

Reply:
Feilmeldingen uten cast: function jsonb_extract_path_text(jsonb, bytea) does not exist. Så av en eller anna grunn så skjønner den ikkje at det er tekst uten at vi forteller den det.